package/libsolv.spec.in

```diff
@@ -190,83 +165,80 @@ Group:          Development/Languages/Python
 Python3 bindings for libsolv.
 
 %package -n perl-solv
-Requires:       perl = %{perl_version}
 Summary:        Perl bindings for the libsolv library
 Group:          Development/Languages/Perl
+Requires:       perl = %{perl_version}
 
 %description -n perl-solv
 Perl bindings for libsolv.
 
 %prep
-%setup -n libsolv-%{version}
+%setup -q
 
 %build
 export CFLAGS="%{optflags}"
 export CXXFLAGS="$CFLAGS"
 
 CMAKE_FLAGS=
-%if 0%{?fedora} || 0%{?rhel} >= 6
-CMAKE_FLAGS="-DFEDORA=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+%if 0%{?fedora} || 0%{?rhel}
+CMAKE_FLAGS="-DFEDORA=1"
 %endif
 %if 0%{?mageia}
-CMAKE_FLAGS="-DMAGEIA=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+CMAKE_FLAGS="-DMAGEIA=1"
 %endif
 %if 0%{?suse_version}
-CMAKE_FLAGS="-DSUSE=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+CMAKE_FLAGS="-DSUSE=1"
 %endif
 
-cmake	$CMAKE_FLAGS \
-	-DCMAKE_INSTALL_PREFIX=%{_prefix} \
-	-DLIB=%{_lib} \
-	-DCMAKE_VERBOSE_MAKEFILE=TRUE \
-	-DCMAKE_BUILD_TYPE=RelWithDebInfo \
+# SHARE/MODULE flags in SUSE enforce no-undefined which is not applicable for
+# bindings here
+%cmake $CMAKE_FLAGS \
+	-DCMAKE_MODULE_LINKER_FLAGS="-Wl,--as-needed -Wl,-z,now" \
+	-DCMAKE_SHARED_LINKER_FLAGS="-Wl,--as-needed -Wl,-z,now" \
 	-DWITH_LIBXML2=1 \
+	-DENABLE_APPDATA=1 \
+	-DENABLE_COMPS=1 \
 	%{?with_enable_static:-DENABLE_STATIC=1} \
 	%{?with_disable_shared:-DDISABLE_SHARED=1} \
-	%{?with_perl_binding:-DENABLE_PERL=1} \
-	%{?with_python_binding:-DENABLE_PYTHON=1} \
-	%{?with_python3_binding:-DENABLE_PYTHON3=1} \
-	%{?with_ruby_binding:-DENABLE_RUBY=1} \
+	%{?with_perl:-DENABLE_PERL=1} \
+	%{?with_python:-DENABLE_PYTHON=1} \
+	%{?with_python3:-DENABLE_PYTHON3=1} \
+	%{?with_ruby:-DENABLE_RUBY=1} \
 	%{?with_bz2:-DENABLE_BZIP2_COMPRESSION=1} \
 	%{?with_xz:-DENABLE_LZMA_COMPRESSION=1} \
 	%{?with_richdeps:-DENABLE_COMPLEX_DEPS=1} \
 	%{?with_zypp:-DENABLE_SUSEREPO=1 -DENABLE_HELIXREPO=1} \
-	-DUSE_VENDORDIRS=1 \
-	-DCMAKE_SKIP_RPATH=1
+	-DUSE_VENDORDIRS=1
 make %{?_smp_mflags}
 
 %install
-make DESTDIR=%{buildroot} install
-%if 0%{?suse_version}
-%if %{with python_binding}
-pushd %{buildroot}/%{python_sitearch}
-python %py_libdir/py_compile.py *.py
-python -O %py_libdir/py_compile.py *.py
-popd
+%cmake_install
+
+%if %{with python2}
+%py_compile %{buildroot}/%{python_sitearch}
```

Comment:
Only SUSE systems don't automatically byte-compile.

Also, use `%python2_sitearch` instead of `%python_sitearch`.

Add a compatibility definition for older SUSE systems:
```
%{!?python2_sitearch:%global %python2_sitearch %{python_sitearch}}
```

Has the `%py_compile` macro been renamed to `%py2_compile` yet?
 
(also I don't know if `%pyX_compile` exists on non-SUSE anyway)

Reply:
There is no %py2_compile, also python2_sitearch is not declared on older suse systems so why should I pollute the spec just for external compatibility.

Reply:
It exists at least for Leap 42.3?